browser/static/js/actions.js

```diff
@@ -0,0 +1,179 @@
+class ToggleEdit extends Action {
+
+    do() {
+      edit_mode = !edit_mode;
+    }
+
+    undo() {
+      this.do()
+    }
+
+    redo() {
+      this.do()
+    }
+  }
+
+class ToggleHighlight extends Action {
+
+  do() {
+    current_highlight = !current_highlight;
+  }
+
+  undo() {
+    this.do()
+  }
+
+  redo() {
+    this.do()
+  }
+}
+
+/** Action to change the viewed frame. */
+class ChangeFrame extends Action {
```

Comment:
I don't think that this should be logic inside the ChangeFrame class, because an Action can't tell the history if it was actually performed or not. Instead, we shouldn't create & perform the action in the first place. I'll add conditions to the key handlers that check that `max_frames > 1`, etc.